dd-trace-ot/src/main/java/datadog/opentracing/PendingTrace.java

```diff
@@ -202,9 +205,12 @@ private void expireReference() {
     if (count == 0) {
       write();
     } else {
-      if (tracer.getPartialFlushMinSpans() > 0 && size() > tracer.getPartialFlushMinSpans()) {
+      if (tracer.getPartialFlushMinSpans() > 0
+          && size() > tracer.getPartialFlushMinSpans()
+          && size() + partiallyWrittenSpanCount.get() < Config.get().getMaxSpansPerTrace()) {
```

Comment:
Depending on their size and rate of accumulation, this will reveal jumbo traces as leaks as opposed to flushing them as invalid ones.  Is this what we want, or would it be better to move the partially written count informed check below to bypass the `tracer.write()`, still decrementing the completed span count?

Reply:
Excellent point.  I will look at it in more depth, again after I feel like the tests are happier/less flaky.